hammer/workers.go

```diff
@@ -201,7 +201,9 @@ func (w *LogWriter) Run(ctx context.Context) {
 		lt := leafTime{queuedAt: time.Now()}
 		index, err := w.writer(ctx, newLeaf)
 		if err != nil {
-			w.errChan <- fmt.Errorf("failed to create request: %v", err)
+			if !errors.Is(ErrRetry, err) {
+				w.errChan <- fmt.Errorf("failed to create request: %w", err)
```

Comment:
I think it deserves a comment to say that as implemented this retry uses a token from the throttle for each attempt. So if we had a throttle of 100 writes per second, but one thread retried 99 times in a second, we'd only have 1 leaf written and no errors propagated down `w.errChan` to explain where all the writes were going.